### PR TITLE
Restructure OpenTelemetry Documentation

### PR DESCRIPTION
**Pull request by brett0000FF**
### What does this PR do? What is the motivation?

Restructure OpenTelemetry documentation into a more logical structure. This will make it easier for readers to find what they are looking for, and it will improve maintenance of these docs as we continue to add more content.

Jira: https://datadoghq.atlassian.net/browse/DOCS-6974
Brainstorming: [FigJam](https://www.figma.com/file/wW2cAzPkZ5dZ8QcaS8stF1/OpenTelemetry-Docs-Restructure?type=whiteboard&node-id=0%3A1&t=UTsGxvpzjyEc7nwk-1).

Preview: https://docs-staging.datadoghq.com/brett0000FF/otel-reorg/opentelemetry/

- Added Terms and Concepts page.
- Split Set up Collector and Exporter into Deployment, Configuration, Management, and Integrations pages.
- Added Schema and Semantics section that organizes related content.
- Added Interoperability with Datadog section that organizes related content.
- New Troubleshooting page.
- Updated Guides to include resources as well (like the Registry).

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

**Comment by julianocosta89** [comment]
Hello @brett0000FF 👋🏽 

I think this re-structure is a good start, IMO we are missing some pages in the docs, but we can definitely work on that after this initial PR.

Just one thing that I'd suggest, not sure if it is doable:

In the `Data Collection` section we go to the page, and we have 2 links to chose from.
Would it be possible to make this page as the final one?
I mean, the user clicks on the `Data Collection` and it is presented with the differences between the 2 options, focusing on the advantages of each one of them and then 2 tabs explaining each case: 
- `OTel Collector`
- `Datadog Agent`

WDYT?

**Comment by brett0000FF** [comment]
> In the `Data Collection` section we go to the page, and we have 2 links to chose from. Would it be possible to make this page as the final one? I mean, the user clicks on the `Data Collection` and it is presented with the differences between the 2 options, focusing on the advantages of each one of them and then 2 tabs explaining each case:
> 
> * `OTel Collector`
> * `Datadog Agent`
> 
> WDYT?
@julianocosta89 - Yeah, I definitely agree with that. I planned to expand on that content to explain the differences, rather than just a simple link. I met with Irene today about this structure, and she's going to suggest some other high-level structure changes. Once we get that sorted out, it'll become more clear where we need to fill in content gaps. Thanks for taking a look!

**Comment by julianocosta89** [comment]
@brett0000FF https://docs-staging.datadoghq.com/brett0000FF/otel-reorg/opentelemetry/interoperability/otlp_delta_temporality/ seems to be failing.

Also, I've mentioned you on the TOC Doc, where I said that we should move `Datadog Exporter` and `OTLP Receiver` to `Configuration` and leave the others in `Integrations`, as those 2 are not integrations.

File: content/en/opentelemetry/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+```yaml
+processors:
+  resourcedetection:
+    detectors: [env, ecs]
+    timeout: 2s
+    override: false
+```
+
+**Note**: ECS Fargate is supported for v0.42+ of the Datadog Exporter.
```
Comment (julianocosta89) [inline content/en/opentelemetry/troubleshooting.md:54]: What does that mean?
The `processor` is a different component of the collector.
It adds resource attributes to the data which will be sent (traces, metrics and logs), depending on how you configure your pipelines.

How come that be supported just for `v0.42+` of the Datadog Exporter, which is a different component of the collector?
Are we dropping the resource attributes if it is an early version?

Comment (brett0000FF) [reply content/en/opentelemetry/troubleshooting.md:54]: Thanks! I'll make an adjustment here. I didn't mean to say Exporter configuration.

I'm not 100% sure about the Exporter version. I know that support for Fargate was added here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6670. This was something I grabbed from Support. Maybe I'll remove the note for now as it may just add more noise to the section.

File: content/en/opentelemetry/collector_exporter/management.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
```
Comment (julianocosta89) [inline content/en/opentelemetry/collector_exporter/management.md:1]: Do we actually need this page?
This is great, but it is not related to Datadog, and we are not teaching the user how to do it.
They still need to go to https://opentelemetry.io/docs/collector/management/#opamp in order to learn more.

I'd get rid of this section.

File: content/en/opentelemetry/otel_terms.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+|------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Telemetry                    | The collection of metrics, logs, and traces that provide observations about the behaviors and performance of applications and infrastructure.                                                                                                                             |
+| [OpenTelemetry Collector][1] | A vendor-agnostic agent process for collecting and exporting telemetry data emitted by many processes. It can be configured to receive, process, and export telemetry to one or multiple destinations including storage backends and analysis tools.                      |
```
Comment (julianocosta89) [inline content/en/opentelemetry/otel_terms.md:21]:
```suggestion
| [OpenTelemetry Collector][1] | A vendor-agnostic way to process, collect and export telemetry data emitted by many processes. It can be configured to receive, process, and export telemetry to one or multiple destinations including storage backends and analysis tools.                      |
```

Comment (julianocosta89) [reply content/en/opentelemetry/otel_terms.md:21]: The collector is not actually an agent.
It can be deployed as such, but I'd rephrase it.

File: content/en/opentelemetry/collector_exporter/management.md (hunk 2 above)
Comment (brett0000FF) [reply content/en/opentelemetry/collector_exporter/management.md:1]: Sounds good. I'll remove this page and instead put a link to the OTel Management docs in the Configuration further reading. 👍

File: content/en/opentelemetry/collector_exporter/configuration.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+{{< whatsnext desc=" " >}}
+    {{< nextlink href="/opentelemetry/collector_exporter/otel_collector_datadog_exporter/" >}}Datadog Exporter{{< /nextlink >}}
+    {{< nextlink href="/opentelemetry/collector_exporter/otlp_receiver/" >}}OTLP Receiver{{< /nextlink >}}
+    {{< nextlink href="/opentelemetry/collector_exporter/hostname_tagging/" >}}Hostname and Tags{{< /nextlink >}}
+    {{< nextlink href="/opentelemetry/collector_exporter/collector_batch_memory/" >}}Batch and Memory Settings{{< /nextlink >}}
+    {{< nextlink href="/opentelemetry/collector_exporter/docker_metrics/" >}}Docker Metrics{{< /nextlink >}}
+    {{< nextlink href="/opentelemetry/collector_exporter/host_metrics/" >}}Host Metrics{{< /nextlink >}}
+    {{< nextlink href="/opentelemetry/collector_exporter/trace_metrics/" >}}Trace Metrics{{< /nextlink >}}
+    {{< nextlink href="/opentelemetry/collector_exporter/log_collection/" >}}Log Collection{{< /nextlink >}}
+    {{< nextlink href="/opentelemetry/collector_exporter/collector_health_metrics/" >}}Collector Health Metrics{{< /nextlink >}}
```
Comment (julianocosta89) [inline content/en/opentelemetry/collector_exporter/configuration.md:30]: Those are already listed on `Integrations`

**Review by julianocosta89 — APPROVED** [review]
1 minor suggestion, but LGTM 🚀

File: content/en/opentelemetry/collector_exporter/configuration.md (hunk 4 above)
Comment (brett0000FF) [reply content/en/opentelemetry/collector_exporter/configuration.md]: Good catch! It was complicated moving all of this stuff around. 😆

File: content/en/opentelemetry/schema_semantics/_index.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+kind: documentation
+further_reading:
+- link: "/opentelemetry/interoperability/"
+  tag: "Documentation"
+  text: "Interoperability with Datadog"
+---
+
+## Overview
+
+OpenTelemetry uses of semantic conventions to specify names for different types of data. Semantic conventions define common names and values for telemetry data to promote consistency across systems.
```
Comment (buraizu) [inline content/en/opentelemetry/schema_semantics/_index.md:12]:
```suggestion
OpenTelemetry uses semantic conventions to specify names for different types of data. Semantic conventions define common names and values for telemetry data to promote consistency across systems.
```

File: content/en/opentelemetry/otel_terms.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+| [OpenTelemetry Collector][1] | A vendor-agnostic implementation for collecting and exporting telemetry data emitted by many processes. It can be configured to receive, process, and export telemetry to one or multiple destinations including storage backends and analysis tools.                      |
+| [Datadog Exporter][2]        | A component that lets you to forward trace, metric, and logs data from OpenTelemetry SDKs to Datadog.                                                                                                                                                                     |
+| [OTLP Receiver][3]           | A component within the OpenTelemetry Collector responsible for accepting telemetry data in the OpenTelemetry Protocol (OTLP) format. OTLP is the native protocol for OpenTelemetry, designed for transferring telemetry data transfer between the SDKs and the Collector. |
```
Comment (buraizu) [inline content/en/opentelemetry/otel_terms.md:23]:
```suggestion
| [OTLP Receiver][3]           | A component within the OpenTelemetry Collector responsible for accepting telemetry data in the OpenTelemetry Protocol (OTLP) format. OTLP is the native protocol for OpenTelemetry, designed for transferring telemetry data between the SDKs and the Collector. |
```

**Review by buraizu — APPROVED** [review]
Looks good; approving with some minor feedback!
